components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListener.java

```diff
@@ -178,48 +182,75 @@ public boolean doPreSetUserClaimValueWithID(String userID, String claimURI, Stri
             // Validate whether claim update request is for a provisioned user.
             validateClaimUpdate(getUsernameFromUserID(userID, userStoreManager));
         }
-        // Validate dob value against the regex.
-        validateDateOfBirthClaimValue(claimURI, claimValue, userStoreManager);
+        // Validate dob and mobile number value against the regex.
+        validateClaimValue(claimURI, claimValue, userStoreManager);
         // Validate if the groups are updated.
         validateUserGroupClaim(userID, claimURI, claimValue, userStoreManager);
         return true;
     }
 
-    private void validateDateOfBirthClaimValue(String claimURI, String claimValue, UserStoreManager userStoreManager)
+    private void validateClaimValue(String claimURI, String claimValue, UserStoreManager userStoreManager)
             throws UserStoreException {
 
+        String tenantDomain = IdentityTenantUtil.getTenantDomain(userStoreManager.getTenantId());
         if (StringUtils.equalsIgnoreCase(DATE_OF_BIRTH_LOCAL_CLAIM, claimURI)) {
-            String tenantDomain = IdentityTenantUtil.getTenantDomain(userStoreManager.getTenantId());
-            Map<String, String> dateOfBirthClaimProperties = getDateOfBirthClaimProperties(tenantDomain);
-            String dateOfBirthRegex = dateOfBirthClaimProperties.get(PROP_REG_EX);
-            if (StringUtils.isEmpty(dateOfBirthRegex)) {
-                dateOfBirthRegex = DATE_OF_BIRTH_REGEX;
+            validateClaimValueForRegex(claimURI, claimValue, tenantDomain, DATE_OF_BIRTH_REGEX,
+                    DOB_REG_EX_VALIDATION_DEFAULT_ERROR);
+        } else if (StringUtils.equalsIgnoreCase(MOBILE_LOCAL_CLAIM, claimURI)) {
+            validateClaimValueForRegex(claimURI, claimValue, tenantDomain, MOBILE_REGEX,
+                    MOBILE_REG_EX_VALIDATION_DEFAULT_ERROR);
+        }
+    }
+
+    private void validateClaimValueForRegex(String claimURI, String claimValue, String tenantDomain,
```

Comment:
Here also better to add a method comment.

Reply:
fixed in e09571c